src/drivers/driver.ts

```diff
@@ -614,10 +615,64 @@ export abstract class CMakeDriver implements vscode.Disposable {
     await this.doPreCleanConfigure();
     this.configRunning = false;
 
-    return this.configure(extra_args, consumer);
+    return this.configure(trigger, extra_args, consumer);
   }
 
-  async configure(extra_args: string[], consumer?: proc.OutputConsumer, withoutCmakeSettings:boolean = false): Promise<number> {
+  async testCompilerVersion(program: string, cwd: string, arg: string | undefined, regexp: RegExp): Promise<string> {
+    const args = [];
+    if (arg) {
+      args.push(arg);
+    }
+    const child = this.executeCommand(program, args, undefined, {silent: true, cwd});
+    try {
+      const result = await child.result;
+      log.debug(localize('command.version.test.return.code', 'Command version test return code {0}', nullableValueToString(result.retc)));
+      // cl.exe version information is printed on stderr when no source files are given
+      const versionLine = (program === "cl") ? result.stderr : result.stdout; //.split("\\n")[0];
+      const match = regexp.exec(versionLine);
+      return match ? match[1] : "error";
+    } catch (e) {
+      const e2: NodeJS.ErrnoException = e;
+      log.debug(localize('compiler.version.return.code', 'Compiler version test return code {0}', nullableValueToString(e2.code)));
+      if (e2.code === 'error') {
+        return e2.code;
+      }
+      throw e;
```

Comment:
This function should not be allowed to throw because this is just telemetry.  It needs to return a string.  If there's an error, then return an error string that is easy for telemetry to filter on (i.e. don't make it different per compiler).